sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -139,13 +142,20 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
                 return new ArraySegment<byte>();
             }
 
-            using var memStream = new MemoryStream(StreamBufferSizeInBytes);
-            stream.CopyTo(memStream, StreamBufferSizeInBytes);
-
-            return new ArraySegment<byte>(memStream.ToArray());
+            switch (stream)
+            {
+                case BufferListStream bufferListStream:
```

Comment:
I'm a bit uneasy about this as it isn't exposed as part of ToStream. Right now the code would always hit this branch, but if the implementation were to change in the AMQP lib, it would start hitting the default case which hasn't been fully tested.

Reply:
I don't fully understand. The default case is the existing code. Amqp lib in place uses BufferedListStream alll over the place. This lib also uses it internally. So if the lib switches away from it it is anyway a major breaking change. So the default case is fully tested. Can you elaborate?

Reply:
I'm saying that if we make this change, the "default" branch will no longer be hit - we will only be hitting the BufferListStream case. I agree that it is currently being tested, but with this change it will no longer be tested going forward.

The places where we use BufferListStream are in APIs that actually require a BufferListStream. I don't think there are any places where we use it when the API expects or returns a Stream, and we just use the fact that we know it is actually implemented with a BufferListStream.

Reply:
Then the best way to move forward would be to throw on default. Since we take a more or less exact dependency to a specific version of MS Amqp and release new versions of the SDK when that dependency is updated any potential problem would be caught anyway.

I know this code is a liskov violator but I think that is a reasonable tradeoff given the constraints this lib operates on

Reply:
I'd agree that throwing on default would be better than just leaving in default, but I also think it would be even better to just not do this change 😄 
If others disagree though, then I'm open to reconsidering.
/cc @jsquire @pakrym

Reply:
Maybe we can see if the AmqpLib can add a ToBufferStream method..

Reply:
Hehe. I'm not going to fight for it 🤣 I can easily isolate the other change into a dedicated PR if needed

Reply:
Yeah, that would be great! We can discuss this specific change further, but the other one should be good to go.